### DISCUSSION TITLE
config context input for prefix list

### DISCUSSION THREAD

**Discussion by erikgadioli** [peering-manager/peering-manager#855] [Q&A] [answered]
Could config context be use as input for example generating a prefix list, in a jinja template?

like this, direct session config context:
{
    "prefixes-inet": [
        "5.0.0.0/8",
        "194.0.0.0/19"
    ]
}

If i use this in jinja:
{{ session | context_get_key('prefixes-inet') }}

it results in the following config output ['5.0.0.0/8', '194.0.0.0/19']    

is the some way to make it usable in config like below? Can i iter trough context_get_key('prefixes-inet') ?

prefix-list as65001-v4 {
    5.0.0.0/8;
    194.0.0.0/19;
}

This example is a prefix list, could be anything ofcource. 

Tnx and kind regards, Erik

**Comment by gmazoyer** [comment, marked as answer]
`context_get_key` returns whatever type the variable is of. Config contexts are Python dictionaries so by getting a value of a given key you will be able to use it as it is intended to. So, in your case, you can do something like:

```jinja2
{% for prefix in session | context_get_key('prefixes-inet') %}
{{ prefix }};
{% endfor %}
```

**Reply by erikgadioli** [reply]
wow awesome, that is powerfull!
